dace/transformation/subgraph/helpers.py

```diff
@@ -77,70 +82,247 @@ def find_reassignment(maps: List[nodes.Map],
+    subset_set = set(
+        subsets.Range.from_indices(s) if isinstance(s, subsets.Indices) else s
+        for s in subset_list)
+    while True:
+        for sa, sb in itertools.product(subset_set, subset_set):
+            if sa is sb:
+                continue
+            if sa.covers(sb):
+                subset_set.remove(sb)
+                break
+            elif sb.covers(sa):
+                subset_set.remove(sa)
+                break
+            elif are_subsets_contiguous(sa, sb, dim):
+                subset_set.remove(sa)
+                subset_set.remove(sb)
+                subset_set.add(subsets.bounding_box_union(sa, sb))
+                break
+        else:  # No modification performed
+            break
+    return subset_set
+
+
+def deduplicate(sdfg, graph, map_entry, out_connector, edges):
+    ''' applies Deduplication to ALL edges coming from the same
+        out_connector specified in out_connector.
+        Suitable after consolidating edges at the entry node.
+        WARNING: This is not guaranteed to be deterministic
+    '''
+    # Steps:
+    # 1. Find unique subsets
+    # 2. Find sets of contiguous subsets
+    # 3. Create transients for subsets
+    # 4. Redirect edges through new transients
+
+    # only connector we are interested in
+    conn = out_connector
+
+    # Get original data descriptor
+    edge0 = next(iter(edges))
+    dname = edge0.data.data
+    desc = sdfg.arrays[edge0.data.data]
+
+    # Get unique subsets
+    unique_subsets = set(e.data.subset for e in edges)
+
+    # Find largest contiguous subsets
+    contiguous_subsets = find_contiguous_subsets(unique_subsets)
+    #print("Subsets:", contiguous_subsets)
+
+    # Map original edges to subsets
+    edge_mapping = defaultdict(list)
+    for e in edges:
+        for ind, subset in enumerate(contiguous_subsets):
+            if subset.covers(e.data.subset):
+                edge_mapping[ind].append(e)
+                break
+        else:
+            raise ValueError("Failed to find contiguous subset for edge %s" %
+                             e.data)
+
+    # Create transients for subsets and redirect edges
+    for ind, subset in enumerate(contiguous_subsets):
+        name, _ = sdfg.add_temp_transient(subset.size(), desc.dtype)
+        anode = graph.add_access(name)
+        graph.add_edge(map_entry, conn, anode, None,
+                       Memlet(data=dname, subset=subset))
+        for e in edge_mapping[ind]:
+            graph.remove_edge(e)
+            new_memlet = copy.deepcopy(e.data)
+            # Offset memlet to match new transient
+            new_memlet.subset.offset(subset, True)
+            new_edge = graph.add_edge(anode, None, e.dst, e.dst_conn,
+                                      new_memlet)
+            # Rename data on memlet
+            for pe in graph.memlet_tree(new_edge):
+                pe.data.data = name
+                pe.data.subset.offset(subset, True)
```

Comment:
All of these methods here should be removed from DeduplicateAccess and these versions called from there to deduplicate the code. Unless something changed in the implementations here?

Reply:
Outfactored to transformations/helpers.py (makes more sense there than in subgraph/helpers.py)
Removed staticmethods from dedup_access.py